### PR TITLE
[FrameworkBundle] Move session configuration to PHP

### LINKED ISSUE #37186: Move internal Container configuration from XML to PHP
In Symfony 6, I we will promote usage of configuration written in PHP instead of YAML. For third-party bundles and core, we should do the same, replacing XML with PHP. Doing so would remove the need for the XML lib for core.

The biggest advantage is auto-completion with any modern IDE without explicit support for Symfony, and probably one less thing to learn (how to configure things in YAML/XML).

Be warned that semantic configuration will stay in YAML for now (as using PHP is much harder there).

#36778 did the work for the Twig bundle as an example.

We now need help from the community to move all the other bundles:

**PLEASE, comment here BEFORE starting to work on something and WAIT for the confirmation to avoid duplicate work.**

- [x] src/Symfony/Bundle/DebugBundle/Resources/config/services.xml (@jschaedl)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/annotations.xml (@magnetik)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/assets.xml (@NguyenTruongLinh)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/cache*.xml (@iamvar)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/collectors.xml (@hvt)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/console.xml (@AhmedRaafat14)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/debug*.xml (@TuxBoy)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/error_renderer.xml (@benji07)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.xml (@tuanminhgp)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/form*.xml (@misekai)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/fragment*.xml (@idetox)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/http*.xml (@IonBazan)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/identity_translator.xml (@smmd)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/lock.xml (@tomasjav)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/mailer*.xml (@instabledesign)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger*.xml (@hyoa)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/mime_type.xml (@GromNaN)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/notifier*.xml (@szepczynski)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/profiling.xml (@hvt)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/property_access.xml (@qneyrat)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/property_info.xml (@qneyrat)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/request.xml (@dangkhoagms)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/routing*.xml (@phamuyentri)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/secrets.xml (@GromNaN)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/security_csrf.xml (@Noweh)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/serializer.xml (@mamontovdmitriy)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/services.xml (@rvanlaak)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/session.xml (@cocorambo)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/ssi.xml (@50bhan)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/test.xml (@GaryPEGEOT)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/translation*.xml (@malteschlueter)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/validator*.xml (@simivar)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/web*.xml (@ck-developer)
- [x] src/Symfony/Bundle/FrameworkBundle/Resources/config/workflow.xml (@lyrixx)
- [x] src/Symfony/Bundle/SecurityBundle/Resources/config/collectors.xml (@JudicaelR)
- [x] src/Symfony/Bundle/SecurityBundle/Resources/config/console.xml (@JudicaelR)
- [x] src/Symfony/Bundle/SecurityBundle/Resources/config/guard.xml (@JudicaelR)
- [x] src/Symfony/Bundle/SecurityBundle/Resources/config/security*.xml (@qneyrat)
- [x] src/Symfony/Bundle/SecurityBundle/Resources/config/templating_twig.xml (@ck-developer)
- [x] src/Symfony/Bundle/WebProfilerBundle/Resources/config/*.xml (@jschaedl)

### PR DESCRIPTION
**Pull request by cocorambo**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #37186
| License       | MIT

Move session configuration file to php

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/session.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                service('session.storage.metadata_bag'),
+            ])
+
+        ->set('session.flash_bag', FlashBag::class)
+            ->factory([service('session'), 'getFlashBag'])
+            ->deprecate(
+                'symfony/framework-bundle',
+                '5.1',
+                'The "session.flash_bag" service is deprecated, use "$session->getFlashBag()" instead.',
+            )
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/session.php:74]: Let's consistently put all these arguments in one line (same below)

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/session.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+        ->set('session.marshaller', IdentityMarshaller::class)
+
+        ->set('session.marshalling_handler', MarshallingSessionHandler::class)
+            ->args([
+                service('session.marshalling_handler.inner'),
+                service('session.marshaller'),
+            ])
+    ;
+};
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/session.php:125]: missing newline

**Review by yceruto — CHANGES_REQUESTED** [review]

**Comment by yceruto** [comment]
Please, check the fabbot report, there are other issues to solve: https://fabbot.io/report/symfony/symfony/37217/916dcec7c18845a635e11bb1bf796bbe426ab3f3

**Review by yceruto — APPROVED** [review]
LGTM, thanks!

**Comment by simivar** [comment]
Shouldn't you remove old `session.xml` file?

**Review by simivar — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/session.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            ->public()
+            ->args([
+                service('session.storage'),
+                null, // AttributeBagInterface
+                null, // FlashBagInterface
+                [service('session_listener'), 'onSessionUsage'],
+            ])
+        ->alias(SessionInterface::class, 'session')
+        ->alias(SessionStorageInterface::class, 'session.storage')
+        ->alias('SessionHandlerInterface', 'session.handler')
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/session.php:46]: `\SessionHandlerInterface::class`

Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/session.php:46]: `\SessionHandlerInterface::class` as it's a php class from the global namespace

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/session.xml [hunk 4]
```diff
@@ -1,95 +0,0 @@
...
-            </argument>
-            <argument>%kernel.debug%</argument>
-        </service>
-
-        <!-- for BC -->
-        <service id="session.storage.filesystem" alias="session.storage.mock_file" />
-
-        <service id="session.marshaller" class="Symfony\Component\HttpFoundation\Session\Storage\Handler\IdentityMarshaller" />
-
-        <service id="session.marshalling_handler" decorates="session.handler" class="Symfony\Component\HttpFoundation\Session\Storage\Handler\MarshallingSessionHandler">
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/session.xml:90]: `decorates="session.handler"` seems to be missing in the php config

**Review by Tobion — CHANGES_REQUESTED** [review]

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @cocorambo.

**Comment by rvanlaak** [comment]
@Tobion merging this PR did close #37186 because of the description.

FYI same would happen for #37204 , #37208 and #37222

**Pull request merged**